homeassistant/components/fujitsu_hvac/climate.py

```diff
@@ -0,0 +1,188 @@
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_HALVES, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    FUJI_TO_HA_FAN,
+    FUJI_TO_HA_HVAC,
+    FUJI_TO_HA_SWING,
+    HA_TO_FUJI_FAN,
+    HA_TO_FUJI_HVAC,
+    HA_TO_FUJI_SWING,
+)
+from .coordinator import FujitsuHVACCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up one Fujitsu HVAC device."""
+    api = hass.data[DOMAIN][entry.entry_id]
+    coordinator = FujitsuHVACCoordinator(hass, api)
+    await coordinator.async_config_entry_first_refresh()
+
+    async_add_entities(
+        [FujitsuHVACDevice(coordinator, dev) for dev in coordinator.data.values()]
+    )
+
+
+class FujitsuHVACDevice(CoordinatorEntity, ClimateEntity):
+    """Represent a Fujitsu HVAC device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_precision = PRECISION_HALVES
+    _attr_target_temperature_step = 0.5
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _enable_turn_on_off_backwards_compatibility: bool = False
+
+    def __init__(self, coordinator: FujitsuHVACCoordinator, dev: FujitsuHVAC) -> None:
+        """Set up static attributes."""
+        super().__init__(coordinator, context=dev.device_serial_number)
+        self._dev = dev
+
+        self._attr_unique_id = dev.device_serial_number
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, dev.device_serial_number)},
+            name=dev.device_name,
+            manufacturer="Fujitsu",
+            model=dev.property_values["model_name"],
+            serial_number=dev.device_serial_number,
+            sw_version=self._dev.property_values["mcu_firmware_version"],
+        )
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self._dev = self.coordinator.data[self.coordinator_context]
+        self.async_write_ha_state()
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        return FUJI_TO_HA_FAN.get(self._dev.fan_speed)
+
+    @property
+    def fan_modes(self) -> list[str]:
+        """Return the list of available fan modes.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        return [
+            FUJI_TO_HA_FAN[mode]
+            for mode in self._dev.supported_fan_speeds
+            if mode in FUJI_TO_HA_FAN
+        ]
```

Comment:
Why don't we iterate over the `FUJI_TO_HA_FAN` and check if the mode is in the device?

Reply:
What would be the advantage of doing it this way vs the way I am doing it today ?

Reply:
What list is longer?

Reply:
FUJI_TO_HA_FAN since it has all possible fan modes.